### DISCUSSION TITLE
Having trouble importing CSV files into DuckDB-Wasm

### DISCUSSION THREAD

**Discussion by yilan007** [duckdb/duckdb-wasm#704] [Q&A] [answered]
I am trying to call `insertCSVFromPath(conn: number, path: string, option: CSVInsertOptions)` to import a csv file into DuckDB-Wasm. In the function signature, what is `conn` referring to? Here is how I call it. I don't know what `conn` is referring to, so I just randomly pick a number. 
![Screen Shot 2022-04-12 at 10 28 46 PM](https://user-images.githubusercontent.com/47465194/163087735-2689edff-b953-4ab0-af8d-12742de7f59a.png)
In this case, I got an error as follows:
<img width="542" alt="Screen Shot 2022-04-12 at 10 30 11 PM" src="https://user-images.githubusercontent.com/47465194/163087899-d602026c-db64-4a6a-83e6-85ba5a9ff1a3.png">
 Could anyone help me with it? Thank you very much!

**Comment by domoritz** [comment, marked as answer]
`conn` is a connection. Here is a code snippet from the DuckDB Observable client. So note that you call `insertCSVFromPath` on the connection. 

```js
 async insertCSV(name, buffer, options) {
    const db = await this.db();
    await db.registerFileBuffer(name, new Uint8Array(buffer));
    const conn = await db.connect();
    await conn.insertCSVFromPath(name, { name, schema: "main", ...options });
    await conn.close();
 }
```

**Reply by yilan007** [reply]
Thank you so much! I got confused by the API documentation which said `conn` must be a number. This makes a lot more sense!
